File: c++/1299. Replace Elements with Greatest Element on Right Side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n=arr.size(); 
        int num=arr[n-1];
        arr[n-1]=-1;
        for(int i=n-2; i>=0; i--) {
            int temp = arr[i];
            arr[i] = num;
            if(temp>num) num = temp;
        }
        return arr;
    }
};

File: JS/1299. Replace Elements with Greatest Element on Right Side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    let n = arr.length;
    let num = arr[n-1];
    arr[n-1] = -1;
    
    for(let i=n-2; i>=0; i--) {
        let temp = arr[i];
        arr[i] = num;
        if(temp > num) num = temp;
    }
    
    return arr;
};
